RecoPixelVertexing/PixelTriplets/plugins/gpuPixelDoubletsAlgos.h

```diff
@@ -72,28 +72,23 @@ namespace gpuPixelDoublets {
                         : innerBarrel && std::abs(mes - int(std::abs(dz / dr) * T::dzdrFact + 0.5f)) > T::maxDYPred;
     }
 
-    __device__ __forceinline__ bool clusterCut(H const& hh, int i, int o) const {
-      auto mo = hh.detectorIndex(o);
-      bool outerFwd = (mo >= T::last_barrel_detIndex);
+    __device__ __forceinline__ bool clusterCut(H hh, int i) const {
 
-      if (!outerFwd)
-        return false;
```

Comment:
the removal of the `outerFwd` check is a fix ?

Reply:
This is worth an explanation. More than removing it I moved it below (likewise) or after ("algorithm-wise"). We were seeing crashed in cases of  events with no hits on some layer here
```cpp 
__device__ __forceinline__ bool clusterCut(H hh, int i, int o) const {
const uint32_t mo = hh[o].detectorIndex(); //<-- here
```
because `o` that comes from 

```auto fo = __ldg(phiBinner.begin(hoff));  //first hit on outer for the cluster cut.```  was not set (and returning a random overflowing number) when a layer is empty this is not set. For a reason I don't understand, before all our changes this was not spotted because the `begin` was returning `0` when the layer was empty so it was still wrong but not causing any harm (basically we were taking the first hit on the first layer as outer hit and that ended up to be discarded anyway later). 

So given this `fo` is used only in the `clusterCut` to check if the hit is on a forward layer I've simplified it by checking it directly in 

```cpp
if (doClusterCut && outer > pixelTopology::last_barrel_layer && cuts.clusterCut(hh, i))
``` 

if we have the `outer` layer in the forward. That is a necessary condition to apply the cut.

Reply:
thanks for the explanation !